src/components/dashboard/Dashboard.tsx

```diff
@@ -0,0 +1,80 @@
+import { PostgrestError } from "@supabase/supabase-js";
+import React, { useCallback, useEffect, useState } from "react";
+import { useToast, useAuth } from "../../contexts";
+
+import {useAchievementDetail} from '../../hook/';
+
+const Dashboard = () => {  
+  const {currentUser} =  useAuth();
+  const { userDetails, getUserDetails, setUserDetails} = useAchievementDetail();
+  const { notifyError } = useToast();
+  
+  useEffect( () => {
+  
+    getUserDetails();
+  }, []);
```

Comment:
Suggestion: if there is an external function/object/variable inside useEffect, include these in the dependency array. Use eslint command "npx eslint ." for warnings.